src-built-in/adapters/indexDBAdapter.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * IndexDB Storage Adapter.
+ */
+
+/**
+ * We have a baseStorage model that provides some methods, such as `getCombinedKey`, which will return a nice key to save our value under. Example: `Finsemble:defaultUser:finsemble:activeWorkspace`. That key would hold the value of our activeWorkspace.
+ */
+// var BaseStorage = require("@chartiq/finsemble").models.baseStorage;
+// var Logger = require("@chartiq/finsemble").Clients.Logger;
+// var Dexie = require('dexie');
+
+import finsemble from "@chartiq/finsemble";
+import Dexie from 'dexie';
+var BaseStorage = finsemble.models.baseStorage;
+var Logger = finsemble.Clients.Logger;
+
+
+// var BaseStorage = require("@chartiq/finsemble").models.baseStorage;
+// var Logger = require("@chartiq/finsemble").Clients.Logger;
+// var Dexie = require('dexie');
+
+
+//Because calls to this storage adapter will likely come from many different windows, we will log successes and failures in the central logger.
+Logger.start();
+
+Logger.system.debug("IndexDBAdapter", Dexie);
+console.debug("IndexDBAdapter", Dexie);
+
+var IndexDBAdapter = function (uuid) {
+	BaseStorage.call(this, arguments);
+
+	var db = new Dexie('finsemble');
+	db.version(1).stores({
+		fsbl: `key` //only index the storage keys
+	});
+
+	Logger.system.debug("IndexDBAdapter init");
+	console.debug("IndexDBAdapter init");
+	/**
+	 * Save method.
+	 * @param {object} params
+	 * @param {string} params.topic A topic under which the data should be stored.
+	 * @param {string} params.key The key whose value is being set.
+	 * @param {any} params.value The value being saved.
+	 * @param {function} cb callback to be invoked upon save completion
+	 */
+	this.save = function (params, cb) {
+		Logger.system.debug("saving", params);
+		const combinedKey = this.getCombinedKey(this, params);
+		
+		Dexie.spawn(function*() {
```

Comment:
Why does spawn need a generator?

Reply:
The generator really changes how the function is executed - inside the generator you can use the yield keyword to wait for an async method (promise) to return as if it were synchronous. The generator itself returns a promise so the whole block is async, but you can write nice simple synchronous looking code inside.

All the db.fsbl calls are promise based - see http://dexie.org/docs/Simplify-with-yield